packages/flutter/lib/src/widgets/scroll_controller.dart

```diff
@@ -109,6 +111,7 @@ class ScrollController {
   /// controller will not manipulate the given position.
   void detach(ScrollPosition position) {
     assert(_positions.contains(position));
+    position.removeListener(notifyListeners);
```

Comment:
You don't have to dispose it.  The scroll positions will detach themselves when they go away.  Only people who listen need to either unlisten or dispose.